File: identhub/src/main/java/de/solarisbank/identhub/session/data/verification/phone/VerificationPhoneApi.java
package de.solarisbank.identhub.session.data.verification.phone;

import de.solarisbank.identhub.session.data.TransactionAuthenticationNumber;
import de.solarisbank.identhub.session.data.verification.phone.model.VerificationPhoneResponse;
import io.reactivex.Single;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface VerificationPhoneApi {
    @POST("mobile_number/authorize")
    Single<VerificationPhoneResponse> postAuthorize();

    @POST("mobile_number/confirm")
    Single<VerificationPhoneResponse> postConfirm(@Body final TransactionAuthenticationNumber tan);
}

File: phone/src/main/java/de/solarisbank/identhub/phone/data/VerificationPhoneApi.kt
package de.solarisbank.identhub.phone.data

import de.solarisbank.identhub.phone.model.TransactionAuthenticationNumber
import de.solarisbank.identhub.phone.model.VerificationPhoneResponse
import io.reactivex.Single
import retrofit2.http.Body
import retrofit2.http.POST

interface VerificationPhoneApi {
    @POST("mobile_number/authorize")
    fun postAuthorize(): Single<VerificationPhoneResponse>

    @POST("mobile_number/confirm")
    fun postConfirm(@Body tan: TransactionAuthenticationNumber): Single<VerificationPhoneResponse>
}